addons/account/models/account_report.py

```diff
@@ -352,6 +353,7 @@ def _create_report_expression(self, engine):
             elif engine == 'aggregation' and report_line.aggregation_formula:
                 subformula, formula = None, report_line.aggregation_formula
             else:
+                report_line.expression_ids.filtered(lambda exp: exp.engine == engine and exp.label == 'balance' and not xml_ids.get(exp)).unlink()
```

Comment:
xml_ids contains integers as keys (record ids), so the get() you do here will never return anything :/ => exp.id